minio/minioadmin.py

```diff
@@ -48,9 +48,12 @@ def _run(self, args, multiline=False):
             capture_output=True,
             timeout=self._timeout,
             check=True,
+            text=True,
         )
+        if not proc.stdout:
+            return [] if multiline else {}
         if multiline:
-            return [json.loads(line) for line in proc.stdout.split("\n")]
+            return [json.loads(line) for line in proc.stdout.split("\n") if line]
```

Comment:
No change required in this line.

Reply:
The error I had actually occurs when proc.stdout is not empty, but has an empty `b""` after `split`.

Reply:
Here is an example:

The response:
```
'{"status":"success","policy":"diagnostics","isGroup":false}\n{"status":"success","policy":"readonly","isGroup":false}\n{"status":"success","policy":"readwrite","isGroup":false}\n{"status":"success","policy":"writeonly","isGroup":false}\n'
```
becomes
```python
['{"status":"success","policy":"diagnostics","isGroup":false}',
 '{"status":"success","policy":"readonly","isGroup":false}',
 '{"status":"success","policy":"readwrite","isGroup":false}',
 '{"status":"success","policy":"writeonly","isGroup":false}',
 '']
```
which fails at the very end because the trailing new-line results in empty string.

Reply:
you could also do something like `proc.stdout.strip().split("\n")` to get rid of any leading/trailing whitespace.

Reply:
Use `.splitlines()` instead of `.split("\n")`

Reply:
That still doesn't handle cases where response is `"a\n\nb\nc\n"`, but it's good enough for me.